### DISCUSSION TITLE
Repeated SIGSEGV crashes

### DISCUSSION THREAD

**Discussion by elrac** [itzg/docker-minecraft-server#3015] [General]
My ram was defective. If you are receiving seemingly random SIGSEGV errors check your ram. 

-----
I have been attempting to run a Fabric Minecraft 1.21 server, first in Manjaro linux (Kernel 6.9) then in Debian (Kernel 6.1) and was having periodic SIGSEGV errors. I could sit in a world and wait, and it would crash in any amount of time from 2 minutes to an hour, but it would never stay up for more than an hour if someone was logged in. The exact same setup runs fine in Linus Mint (Kernel 5.15). I think that Linux kernel version 6 introduced an error with linking that breaks some Fabric functions. I managed to fix the issue for myself with the following Dockerfile:

FROM itzg/minecraft-server:latest

ENV LD_BIND_NOW=1
RUN wget https://chriswhocodes.com/hsdis/hsdis-amd64.so -P /opt/java/openjdk/lib

**Comment by itzg** [comment]
Thanks for the information and research. I'll prefer to wait until OpenJDK/Temurin, etc includes a fix upstream.

**Reply by elrac** [reply]
That's fair. If anyone has the same issue they can make the same Dockerfile. Btw, I've been running this container for at least 3 years and it is great, thank you for making and maintaining it.
